tests/components/rfxtrx/test_binary_sensor.py

```diff
@@ -143,21 +156,58 @@ async def test_off_delay(hass, rfxtrx):
     assert state
     assert state.state == "off"
 
-    await _signal_event(hass, "0b1100100118cdea02010f70")
+    await rfxtrx.signal("0b1100100118cdea02010f70")
     state = hass.states.get("binary_sensor.ac_118cdea_2")
     assert state
     assert state.state == "on"
 
-    base_time = utcnow()
-
-    async_fire_time_changed(hass, base_time + timedelta(seconds=4))
-    await hass.async_block_till_done()
+    await timestep(4)
     state = hass.states.get("binary_sensor.ac_118cdea_2")
     assert state
     assert state.state == "on"
 
-    async_fire_time_changed(hass, base_time + timedelta(seconds=6))
-    await hass.async_block_till_done()
+    await timestep(4)
     state = hass.states.get("binary_sensor.ac_118cdea_2")
     assert state
     assert state.state == "off"
+
+
+async def test_panic(hass, rfxtrx_automatic):
+    """Test panic entities."""
+    rfxtrx = rfxtrx_automatic
+
+    entity_id = "binary_sensor.kd101_smoke_detector_a10900_32"
+
+    await rfxtrx.signal(EVENT_SMOKE_DETECTOR_PANIC)
+    assert hass.states.get(entity_id).state == "on"
+    assert hass.states.get(entity_id).attributes.get("device_class") == "smoke"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5872960)